runtime/lua/vim/lsp/util.lua

```diff
@@ -743,6 +743,19 @@ local function bufwinid(bufnr)
   end
 end
 
+local function get_directory_buffers(path)
```

Comment:
might be able to simplify this a lot by using `vim.fs.parents()` to check if the parent of a buffer path matches the deleted directory path.